src/libopensc/card-epass2003.c

```diff
@@ -854,9 +1182,21 @@ encode_apdu(struct sc_card *card, struct sc_apdu *plain, struct sc_apdu *sm,
 		if (0 != construct_le_tlv(plain, apdu_buf, data_tlv_len, le_tlv,
 				     &le_tlv_len, exdata->smtype))
 			return -1;
-
-	if (0 != construct_mac_tlv(card, apdu_buf, data_tlv_len, le_tlv_len, mac_tlv, &mac_tlv_len, exdata->smtype))
-		return -1;
+	
+	if(exdata->bFipsCertification){
+        	if(plain->lc == 0 && apdu_buf[1] == 0x82 && apdu_buf[2] == 0x01){
+            		if(0 != construct_mac_tlv_case1(card, apdu_buf, data_tlv_len, le_tlv_len, mac_tlv, &mac_tlv_len, exdata->smtype))
+                		return -1;
+        	}
+        	else{
+            		if (0 != construct_mac_tlv(card, apdu_buf, data_tlv_len, le_tlv_len, mac_tlv, &mac_tlv_len, exdata->smtype))
+                		return -1;
+        	}
+    	}
+    	else{
```

Comment:
I think this is the last remaining indenting error

Reply:
ok